src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -121,6 +98,56 @@ case class CsvRelation protected[spark] (
     new InterpretedMutableProjection(casts, startSchema)
   }
 
+  private def readCSV(
+     file: RDD[String],
+     header: Seq[String],
+     schemaFields: Seq[StructField],
+     projection: MutableProjection,
+     row: GenericMutableRow) = {
+
+    val dataLines = if (useHeader) {
+      file.mapPartitionsWithIndex({
+        case (partitionIndex, iter) => if (partitionIndex == 0) iter.drop(1) else iter
```

Comment:
What if there are header lines in other files? I think matching against header is safer. Basically we are willing to pay extra cost for safer parsing.